gps/gps.go

```diff
@@ -127,16 +154,18 @@ func (gps *Device) WriteBytes(bytes []byte) {
 
 // validSentence checks if a sentence has been received uncorrupted
 func validSentence(sentence string) error {
-	if len(sentence) < 4 || sentence[0] != '$' || sentence[len(sentence)-3] != '*' {
+	if len(sentence) < 7 || sentence[0] != '$' || sentence[len(sentence)-3] != '*' {
```

Comment:
What are magic indexes here? Add a comment about the structure of the sentence?

Reply:
Added some const and comments.